### PR TITLE
[RNMobile] Gallery - Button

### PR DESCRIPTION
**Pull request by mkevins**
## Description

This PR introduces a `Button` component for the semi-cross-platform Gallery block. The purpose of this component is to provide UI buttons that can be styled for the native implementation of the Gallery block.

This `Button` component is used by the `GalleryImage` component introduced here: https://github.com/WordPress/gutenberg/pull/18155. The web implementation of the Gallery block uses `IconButton`, which, in turn, uses `Button` from `@wordpress/components`. Initially, I attempted to use the mobile implementation of `IconButton`, and then `Button`, but I struggled to find a way to make this work. Our mobile implementation of the `Button` component seemingly lacks a way to style the button for the purpose of the gallery image.

Ideally, `Button` (or `IconButton`) from `@wordpress/components` can be general enough to allow re-use from the `GalleryImage` component, supporting the respective designs, but perhaps it's best to constrain the scope here for the first iteration of mobile Gallery block, as the potential for reuse of `Button` may require a wider set of changes.

The changeset here is used in the related "parent" PR: https://github.com/WordPress/gutenberg/pull/18111, which includes these changes integrated with related changes in other components necessary for the gallery block.

## To test
Test this component via the main draft PR.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-library/src/gallery/gallery-button.native.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		<TouchableOpacity
+			activeOpacity={ 0.7 }
+			accessible={ true }
+			accessibilityLabel={ accessibilityLabel }
+			accessibilityRole={ 'button' }
+			onPress={ onClick }
+			disabled={ isDisabled }
+		>
+			<View style={ buttonStyle }>
+				{ isString( icon ) ?
```
Comment (pinarol) [inline packages/block-library/src/gallery/gallery-button.native.js]: `isString` check is done inside Icon component, it shouldn't be necessary here, in theory...

**Comment by pinarol** [comment]
> This Button component is used by the GalleryImage component introduced here: #18155. The web implementation of the Gallery block uses IconButton, which, in turn, uses Button from @wordpress/components. Initially, I attempted to use the mobile implementation of IconButton, and then Button, but I struggled to find a way to make this work. Our mobile implementation of the Button component seemingly lacks a way to style the button for the purpose of the gallery image.

Right, I think we need more props for Icon positioning/size and we need to allow overriding styles for active and disabled states separately. I believe this can be investigated as a part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/1411 Currently web is using CSS to override button styles but we don't have a way to do the same. It is tough to re-design a component from `@wordpress/components` inside another PR, I believe it deserves its own PR. And it'd be really nice to handle this in a full x-platform way, so I am hopeful about doing this as a part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/1411

File: packages/block-library/src/gallery/gallery-button.native.js (hunk 1 above)
Comment (mkevins) [reply packages/block-library/src/gallery/gallery-button.native.js]: You're quite right about that :smile: thanks for pointing it out. I've removed the `isString` check.

**Comment by mkevins** [comment]
Hi Pinar, thanks for taking a look at this one.
> It is tough to re-design a component from @wordpress/components inside another PR, I believe it deserves its own PR.

I think that makes a lot of sense, especially if we are aiming for a general cross-platform solution there. Thanks for linking to the issue tracking the cross-platform development for Button.

For the scope of this PR (and the related native Gallery work), is it acceptable to have this separate implementation of `Button`, i.e. defer the necessary modifications to the mobile implementation within `@wordpress/components`, and address code reuse in a later Gallery PR when the cross-platform `Button` is ready?

**Comment by pinarol** [comment]
> For the scope of this PR (and the related native Gallery work), is it acceptable to have this separate implementation of Button, i.e. defer the necessary modifications to the mobile implementation within @wordpress/components, and address code reuse in a later Gallery PR when the cross-platform Button is ready?

I think cross-platform Button component will cover this anyway so I just want to avoid re-work, and we want to make it happen not later than ~2 months. Could you list what kind of changes needs to be done in `@wordpress/components/Button` for Gallery to be able to use it?

**Comment by mkevins** [comment]
Merged with `GalleryImage` (which depends on `Button`). Tracked here: https://github.com/WordPress/gutenberg/pull/18155

**Comment by mkevins** [comment]
> Could you list what kind of changes needs to be done in @wordpress/components/Button for Gallery to be able to use it?
I'll add a comment about that here: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1411.

**Pull request merged**